libp2p/multiaddress.nim

```diff
@@ -569,6 +569,28 @@ iterator items*(ma: MultiAddress): MultiAddress =
     res.data.finish()
     yield res
 
+proc contains*(ma: MultiAddress, codec: MultiCodec): bool {.inline.} =
+  ## Returns ``true``, if address with MultiCodec ``codec`` present in
+  ## MultiAddress ``ma``.
+  result = false
+  for item in ma.items:
+    if item.protoCode() == codec:
+      result = true
```

Comment:
why not simply `return true`?

Reply:
I do not know how to answer on your question really. I just do not like to use `return` keyword everywhere.

Reply:
most of the time, I like to avoid `return` altogether - this code can be rewritten as `ma.items.anyIt(it.protoCode() == codec)` which avoids the control-flow problem completely

Reply:
Sorry @arnetheduck, but the code is looks like equal:
```nim
template anyIt*(s, pred: untyped): bool =
  ## Iterates through a container and checks if some item fulfills the
  ## predicate.
  ##
  ## Unlike the `any proc<#any,openArray[T],proc(T)>`_,
  ## the predicate needs to be an expression using
  ## the ``it`` variable for testing, like: ``anyIt("abba", it == 'a')``.
  ##
  ## See also:
  ## * `any proc<#any,openArray[T],proc(T)>`_
  ## * `allIt template<#allIt.t,untyped,untyped>`_
  ##
  runnableExamples:
    let numbers = @[1, 4, 5, 8, 9, 7, 4]
    assert numbers.anyIt(it > 8) == true
    assert numbers.anyIt(it > 9) == false

  var result = false
  for it {.inject.} in items(s):
    if pred:
      result = true
      break
  result
```

Reply:
I've come to realize that `result` has many pitfalls specially when used with exceptions it's really easy to leak resources/memory.

Reply:
oh, the implementation of `allIt` is a loop indeed - but you've removed the boilerplate for loop in _your_ code and made it "atomic" - there's no way to copy-paste it the wrong way, or exit the loop poorly with a misindent or any other number of issues you can have. 

you've also told the story of what you're trying to accomplish succinctly: in this case, you're not interested in looping _really_ - that's just an implementation detail - you want to see if `any` item matches, and that's exactly what you write.

Reply:
but whatever, the question was merely to understand the rationale behind using the loop+break - it's not a blocker for this PR